src/MessagingAttachment/MessagingAttachment.tsx

```diff
@@ -70,6 +70,63 @@ function handleRemoveClick(e, onRemoveAttachment, attachmentID) {
 
 // /// Sub-components, exported icon components to be used by consumers /// //
 
+export type AttachmentMimetype =
+  | "pdf"
+  | "doc"
+  | "docx"
+  | "xls"
+  | "xlsx"
+  | "csv"
+  | "ppt"
+  | "pptx"
+  | "jpg"
+  | "jpeg"
+  | "gif"
+  | "png"
+  | "svg"
+  | "mp4"
+  | "mov"
+  | "wmv"
+  | "mp3"
+  | "wav"
+  | "m4a"
+  | "flv"
+  | "txt"
+  | null;
+
+function mimetypeToIcon(mimetype: AttachmentMimetype | string): AttachmentIconType {
+  const mapMimeToIcon = {
+    pdf: "pdf",
+    doc: "doc",
+    docx: "doc",
+    xls: "xls",
+    xlsx: "xls",
+    csv: "catchall",
+    ppt: "ppt",
+    pptx: "ppt",
+    jpg: "image",
+    jpeg: "image",
+    gif: "image",
+    png: "image",
+    svg: "image",
+    mp4: "audio",
+    mov: "video",
+    wmv: "video",
+    mp3: "audio",
+    wav: "audio",
+    m4a: "audio",
+    flv: "video",
+    txt: "doc",
+    null: "catchall",
+  };
+
+  // TODO: is there a better/cleaner way to do this?
+  if (Object.keys(mapMimeToIcon).includes(mimetype)) {
+    return mapMimeToIcon[mimetype];
+  }
+  return "catchall";
```

Comment:
can probably simplify this to `return mapMimeToIcon[mimetype] || "catchall"`

Reply:
genius 💡